trunk/SUAVE/Methods/Aerodynamics/Airfoil_Panel_Method/airfoil_analysis.py

```diff
@@ -0,0 +1,219 @@
+            break 
+
+    # ---------------------------------------------------------------------
+    # Bottom surface of airfoil 
+    # ---------------------------------------------------------------------
+    # x and y coordinates 
+    x_bot_vals    =  x[:i_stag][::-1]  # flip arrays on bottom surface (measured from stagnation point on bottom surface)
+    y_bot         =  y[:i_stag][::-1]  
+    x_bot         = np.zeros_like(x_bot_vals )  
+    x_bot[1:]     = np.cumsum(np.sqrt((x_bot_vals[1:] - x_bot_vals [:-1])**2 + (y_bot[1:] - y_bot[:-1])**2)) 
+      
+    # flow velocity and pressure of on botton surface 
+    Ve_bot        = -vt[:i_stag][::-1] # negative because the velocity is measured anti-clockwise around surface 
+    Cp_bot        = 1 - Ve_bot**2 
+        
+    # velocity gradients on bottom surface 
+    dVe_bot       = np.zeros_like(x_bot_vals)
+    dVe_bot_temp  = np.diff(Ve_bot)/np.diff(x_bot_vals)
+    dVe_bot[0]    = dVe_bot_temp[0] 
+    a             = x_bot[1:-2] - x_bot[:-3]
+    b             = x_bot[2:-1] - x_bot[:-3]
+    dVe_bot[1:-2] = (b*dVe_bot_temp[:-2] + a*dVe_bot_temp[1:-1])/(a+b) 
+    dVe_bot[-1]   = dVe_bot_temp[-1]  
+    L_bot         = x_bot[-1] # x - location of stagnation point 
+    
+    # laminar boundary layer properties using thwaites method 
+    x_t_bot, theta_t_bot, del_star_t_bot, H_t_bot, cf_t_bot, Re_theta_t_bot, Re_x_t_bot,delta_t_bot= thwaites_method(0.000001, L_bot , Re_L, x_bot, Ve_bot, dVe_bot)
+    
+    # transition location  
+    tr_crit_bot     = Re_theta_t_bot - 1.174*(1 + 224000/Re_x_t_bot)*Re_x_t_bot**0.46                
+    tr_loc_vals     = np.where(tr_crit_bot > 0)[0] 
+    if len(tr_loc_vals) == 0:# no trasition  
+        i_tr_bot =  len(tr_crit_bot) - 1
+    else: # transition 
+        i_tr_bot = tr_loc_vals[0]    
+   
+    x_tr_bot        = x_t_bot[i_tr_bot] 
+    del_star_tr_bot = del_star_t_bot[i_tr_bot] 
+    theta_tr_bot    = theta_t_bot[i_tr_bot]    
+    delta_tr_bot    = delta_t_bot[i_tr_bot] 
+    
+    x_h_bot, theta_h_bot, del_star_h_bot, H_h_bot, cf_h_bot, delta_h_bot = heads_method(delta_tr_bot,theta_tr_bot, del_star_tr_bot, L_bot - x_tr_bot, Re_L, x_bot - x_tr_bot, Ve_bot, dVe_bot)
+    x_bs            = np.concatenate([x_t_bot[:i_tr_bot], x_h_bot + x_tr_bot] )
+    theta_bot       = np.concatenate([theta_t_bot[:i_tr_bot] ,theta_h_bot] )
+    del_star_bot    = np.concatenate([del_star_t_bot[:i_tr_bot] ,del_star_h_bot] )
+    H_bot           = np.concatenate([H_t_bot[:i_tr_bot] ,H_h_bot] )
+    cf_bot          = np.concatenate([cf_t_bot[:i_tr_bot], cf_h_bot] )
+    delta_bot       = np.concatenate([delta_t_bot[:i_tr_bot], delta_h_bot]) 
+    
+    # ---------------------------------------------------------------------
+    # Top surface of airfoil 
+    # ---------------------------------------------------------------------
+    
+    #re-parameterise based on length of boundary for the top surface of the airfoil
```

Comment:
This section is repeated code from the bottom surface calculation above (lines 57-100) with the only difference being the x, y, and vt indices; can you pull this out and make it a supporting function called in both places to reduce repeated code?

Reply:
created a function. issue resolved